translate/storage/properties.py

```diff
@@ -445,6 +509,206 @@ class DialectStringsUtf8(DialectStrings):
     default_encoding = "utf-8"
 
 
+class proppluralunit(base.TranslationUnit):
+    KEY = 'other'
+
+    def __init__(self, source="", personality="java"):
+        """Construct a blank propunit."""
+        self.personality = get_dialect(personality)
+        super(proppluralunit, self).__init__(source)
+        self.units = {}
+        self.name = u""
+
+    @staticmethod
+    def _get_language_mapping(lang):
+        if lang:
+            locale = lang.replace('_', '-').split('-')[0]
+            cldr_mapping = data.plural_tags.get(locale, data.plural_tags['en'])
+            if cldr_mapping:
+                if len(cldr_mapping) > 0:
+                    return cldr_mapping
+                else:
+                    return [proppluralunit.KEY]
+        return None
+
+    def _get_target_mapping(self):
+        cldr_mapping = proppluralunit._get_language_mapping(self._store.targetlanguage)
+        if cldr_mapping:
+            return cldr_mapping
+        else:
+            return self.units.keys()
+
+    def _get_source_mapping(self):
+        cldr_mapping = proppluralunit._get_language_mapping(self._store.sourcelanguage)
+        if cldr_mapping:
+            return cldr_mapping
+        else:
+            return self.units.keys()
+
+    def _get_units(self, mapping):
+        ret = []
+        if len(self.units) > 1:
+            for name in mapping:
+                if name not in self.units:
+                    unit = propunit("", self.personality.name)
+                    unit.name = self.personality.get_key(self.name, name)
+                    self.units[name] = unit
+                ret.append(self.units[name])
+        else:
+            ret.append(self.units[proppluralunit.KEY])
+        return ret
+
+    def _get_strings(self, strings, mapping):
+        ret = []
+        if len(strings) > 1:
+            for i, name in enumerate(mapping):
+                if i < len(strings):
+                    ret.append(strings[i])
+                else:
+                    ret.append(u"")
+        else:
+            ret.append(strings[0])
+        return ret
+
+    def _get_source_unit(self):
+        self._get_units(self._get_source_mapping())  # Generate missing forms
+        return self.units[proppluralunit.KEY]
+
+    def _get_ordered_units(self):
+        # Used for str (GWT order)
+        mapping = self._get_target_mapping()
+        names = []
+        for name in self.personality.get_cldr_names_order():
+            if name in mapping:
+                names.append(name)
+        return self._get_units(names)
+
+    def hasplural(self, key=None):
+        if key is None:
+            return len(self.units) > 1
+        else:
+            return key in self.units
+
+    def settarget(self, text):
+        if isinstance(text, multistring):
+            strings = text.strings
+        elif isinstance(text, list):
+            strings = text
+        else:
+            strings = [text]
```

Comment:
The plural cases (with list or multistring) are not tested at all, can you please add tests for that as well?

Reply:
I will